sdks/python/apache_beam/runners/interactive/pipeline_instrument.py

```diff
@@ -100,6 +100,11 @@ def __init__(self, pipeline, options=None):
     # (Dict[str, AppliedPTransform]).
     self._cached_pcoll_read = {}
 
+    # A dict from PCollections in the runner pipeline instance to their
+    # corresponding PCollections in the user pipeline instance. Populated
+    # after preprocess().
+    self._runner_pcoll_to_user_pcoll = {}
```

Comment:
Why do we need this map? What does it do?

Reply:
It tells us the correlation between PCollections from the runner pipeline instance and PCollections from the user pipeline instance.
When querying the `values()`, it also tells us what PCollections are cached.
You can even build a pipeline fragment from the `values()`.